client/components/product-list.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import ProductListItem from './product-list-item';
+
+export default class ProductList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      productList: [],
+      rowList: []
+    };
+    this.getProducts = this.getProducts.bind(this);
+  }
+
+  getProducts() {
+    fetch('/api/products')
+      .then(response => response.json())
+      .then(data => {
+        const listItemsArray = data.map(currentValue => {
+          return (<ProductListItem key={currentValue.productId} name={currentValue.name} shortDescription={currentValue.shortDescription} price={currentValue.price} image={currentValue.image} />);
+        });
+        this.setState({ productList: listItemsArray });
```

Comment:
It's best not to store the react elements in state. Instead, create the elements during the `render` phase of the component lifecycle. Only the data should be stored in state.